lib/auth/v4/headerAuthCheck.js

```diff
@@ -63,20 +63,32 @@ function check(request, log, data, awsService) {
 
     let timestamp;
     // check request timestamp
-    if (request.headers['x-amz-date']) {
-        // format of x-amz- date is ISO 8601: YYYYMMDDTHHMMSSZ
-        timestamp = request.headers['x-amz-date'];
+    const xAmzDate = request.headers['x-amz-date'];
+    if (xAmzDate) {
+        const xAmzDateArr = xAmzDate.split('T');
+        // check that x-amz- date has the correct format and after epochTime
+        if (xAmzDateArr[0].length === 8 && xAmzDateArr[1].length === 7 &&
```

Comment:
If there is no 'T', you might not have an arr[1] so this will throw.  You should check first that xAmzDateArr has a length of 2.